.github/workflows/preview.yaml

```
@@ -0,0 +1,30 @@
+name: Vercel Preview Deployment
+env:
+  VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
+  VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID }}
+on:
+  push:
+    branches:
+      - preview
+      - preview-*
+jobs:
+  Deploy-Preview:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```

瑣末な指摘で恐縮ですが、actions/checkout の最新バージョンはv4のようですね

```suggestion
      - uses: actions/checkout@v4
```